tests/acceptance/standard/test_hooks.py

```diff
@@ -0,0 +1,101 @@
+import pytest
+from typing import TYPE_CHECKING
+
+from tests.acceptance import run_gitlabform, get_random_name
+
+
+@pytest.fixture(scope="class")
+def urls():
+    first_name = get_random_name("hook")
+    second_name = get_random_name("hook")
+    first_url = f"http://hooks/{first_name}.org"
+    second_url = f"http://hooks/{second_name}.org"
+    return first_url, second_url
+
+
+class TestHooksProcessor:
+    def get_hook_from_url(self, project, url):
+        return next(h for h in project.hooks.list() if h.url == url)
+
+    def test_hooks_create(self, gl, project, urls):
+        target = project.path_with_namespace
+        first_url, second_url = urls
+
+        test_yaml = f"""
+            projects_and_groups:
+              {target}:
+                hooks:
+                  {first_url}:
+                    push_events: false
+                    merge_requests_events: false
+                  {second_url}:
+                    job_events: false
+                    note_events: false
+            """
+
+        run_gitlabform(test_yaml, target)
+
+        first_created_hook = self.get_hook_from_url(project, first_url)
+        second_created_hook = self.get_hook_from_url(project, second_url)
+        if TYPE_CHECKING:
+            assert isinstance(first_created_hook, ProjectHook)
+            assert isinstance(second_created_hook, ProjectHook)
+        assert len(project.hooks.list()) == 2
+        assert (
+            first_created_hook.push_events,
+            first_created_hook.merge_requests_events,
+        ) == (False, False)
+        assert (second_created_hook.job_events, second_created_hook.note_events) == (
+            False,
+            False,
+        )
+
+    def test_hooks_update(self, gl, project, urls):
+        first_url, second_url = urls
+        target = project.path_with_namespace
+        first_hook = self.get_hook_from_url(project, first_url)
+        second_hook = self.get_hook_from_url(project, second_url)
+
+        update_yaml = f"""
+            projects_and_groups:
+              {target}:
+                hooks:
+                  {first_url}:
+                    id: {first_hook.id}
+                    push_events: true
+                    merge_requests_events: true
+            """
```

Comment:
In this "update hooks" test, let's add/remove events too.

Also, let's add the 2nd hooks config as we discussed in another comment/thread. You're asserting that hook's existence below already, which is good. It's just might not be clear to others why that hook should still exist.

Reply:
Question: updating uses a PUT, not a PATCH. It's probably obvious, but it's maybe worth emphasising here? 
I think one could create a patch update, with python-gitlab (the UpdateMixin has a _get_update_method that can be set), but is it useful?

Reply:
> When changing a hook we are effectively Patching (I'm updating the hook attribute dictionary, not overwriting it completely, before the PUT).

When you call the `project.hooks.update` method from python-gitlab library, is that a `PATCH` request to gitlab api or a `PUT` request.

I was also thinking that before you call `project.hooks.update`, can we check if there's any difference between what we got from configuration vs what is already configured in GitLab? If they are same, then we don't even need to call `project.hooks.update`, right? For now it's probably fine as is too.

> Enforce true means to strictly implement only what is in the config file

Yes, that's what the `enforce: true` under `hooks` would do if it was implemented. Delete all other hooks in the project but only keep/update what's in the config file.

Reply:
sorry, had not seen this reply. When calling update it does a PUT, but in principle one could use either (the update mixin has a "get update method", meaning one can set it to patch or put, default is put). 
I'll add the check before updating.

Reply:
Thank you so much for the update.